src/vs/workbench/services/editor/browser/editorService.ts

```diff
@@ -515,30 +515,34 @@ export class EditorService extends Disposable implements EditorServiceImpl {
 	//#region openEditor()
 
 	openEditor(editor: IEditorInput, options?: IEditorOptions, group?: OpenInEditorGroup): Promise<IEditorPane | undefined>;
-	openEditor(editor: IResourceEditorInput | IUntitledTextResourceEditorInput, group?: OpenInEditorGroup): Promise<ITextEditorPane | undefined>;
+	openEditor(editor: IResourceEditorInput, group?: OpenInEditorGroup): Promise<IEditorPane | undefined>;
+	openEditor(editor: ITextResourceEditorInput | IUntitledTextResourceEditorInput, group?: OpenInEditorGroup): Promise<ITextEditorPane | undefined>;
 	openEditor(editor: IResourceDiffEditorInput, group?: OpenInEditorGroup): Promise<ITextDiffEditorPane | undefined>;
 	async openEditor(editor: IEditorInput | IUntypedEditorInput, optionsOrPreferredGroup?: IEditorOptions | OpenInEditorGroup, preferredGroup?: OpenInEditorGroup): Promise<IEditorPane | undefined> {
 		let typedEditor: IEditorInput | undefined = undefined;
+		let options = isEditorInput(editor) ? optionsOrPreferredGroup as IEditorOptions : editor.options;
 		let group: IEditorGroup | undefined = undefined;
+		let activation: EditorActivation | undefined = undefined;
+
+		if (isOpenInEditorGroup(optionsOrPreferredGroup)) {
+			preferredGroup = optionsOrPreferredGroup;
+		}
 
 		// Resolve override unless disabled
-		let options = isEditorInput(editor) ? (optionsOrPreferredGroup as IEditorOptions) : editor.options;
 		if (options?.override !== EditorOverride.DISABLED) {
-			const [overrideResult, resolvedGroup, resolvedOptions] = await this.doResolveEditor(
-				isEditorInput(editor) ? { editor, options } : editor,
-				isOpenInEditorGroup(optionsOrPreferredGroup) ? optionsOrPreferredGroup : preferredGroup
-			);
-
-			group = resolvedGroup;
-			options = resolvedOptions;
+			const [resolvedEditor, resolvedGroup, resolvedActivation] = await this.doResolveEditor(isEditorInput(editor) ? { editor, options } : editor, preferredGroup);
 
-			if (overrideResult === OverrideStatus.ABORT) {
+			if (resolvedEditor === OverrideStatus.ABORT) {
 				return; // skip editor if override is aborted
 			}
 
-			if (overrideResult !== OverrideStatus.NONE) {
-				typedEditor = overrideResult.editor;
-				options = overrideResult.options ?? options;
+			group = resolvedGroup;
+			activation = resolvedActivation;
+
+			// We resolved an editor to use
+			if (isEditorInputWithOptions(resolvedEditor)) {
+				typedEditor = resolvedEditor.editor;
+				options = resolvedEditor.options;
```

Comment:
This was previously 

```ts
options = overrideResult.options ?? options;
```

But I felt it weird to not just take the options from the `doResolveEditor` call because if an editor is resolved, the options should be fine.